src/bundle/Resources/public/js/scripts/admin.location.visibility.js

```diff
@@ -1,12 +1,36 @@
-(function (global, doc) {
-    const updateVisibility = event => {
-        event.target.closest('.ez-checkbox-icon').classList.toggle('is-checked');
+(function(global, doc) {
+    const SELECTOR_VISIBILITY_CHECKBOXES = '#ez-tab-location-view-locations .ez-checkbox-icon__checkbox';
+    const fetchInit = {
+        credentials: 'same-origin',
+    };
+    const visibilityCheckboxes = [...doc.querySelectorAll(SELECTOR_VISIBILITY_CHECKBOXES)];
 
-        doc.querySelector('#location_update_visibility_data_location').value = event.target.value;
-        doc.querySelector('#location_update_visibility_data_hidden').checked = !event.target.checked;
+    const onVisibilityUpdated = (event) => {
+        const { target } = event;
+        const { checked: isHidden } = target;
 
-        doc.querySelector('form[name="location_update_visibility_data"]').submit();
-    }
+        target.closest('.ez-checkbox-icon').classList.toggle('is-checked', isHidden);
+    };
+    const handleUpdateError = global.eZ.services.notification.showErrorNotification;
+    const handleUpdateResponse = (event) => (response) => {
+        if (response.status === 200) {
+            onVisibilityUpdated(event);
+            return;
+        }
+
+        throw new Error(response.statusText);
+    };
+    const updateVisibility = (event) => {
+        const { value: locationId, checked: isVisible } = event.target;
+        const isHiddenNumberValue = !isVisible ? 1 : 0;
+        const changeVisibilityLink = window.Routing.generate('ezplatform.location.hide', { locationId, hidden: isHiddenNumberValue });
 
-    [...doc.querySelectorAll('.ez-checkbox-icon__checkbox')].forEach(checkbox => checkbox.addEventListener('change', updateVisibility, false));
+        fetch(changeVisibilityLink, fetchInit)
+            .then(handleUpdateResponse(event))
+            .catch(handleUpdateError);
+    };
+
+    for (const checkbox of visibilityCheckboxes) {
```

Comment:
Why not old fashion `forEach`? :)

Reply:
As discussed in private. There is no point to avoid side effects in this case and with `forEach` the code would be simpler.